File: bukkit/src/main/java/io/github/eddiediamondfire/economyplus/vault/EPVaultHook.java
package io.github.eddiediamondfire.economyplus.vault;

import io.github.eddiediamondfire.economyplus.EconomyPlus;
import net.milkbowl.vault.economy.Economy;
import org.bukkit.plugin.ServicePriority;
import org.bukkit.plugin.ServicesManager;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class EPVaultHook {

    private final EconomyPlus plugin;
    private ServicesManager servicesManager = null;
    private final Logger logger;
    private Economy economyProvider = null;
    public EPVaultHook(EconomyPlus plugin){
        this.plugin = plugin;
        logger = plugin.getCustomLogger();
    }

    public void hookVaultIntegration(){
        if(economyProvider == null){
            economyProvider = new EconomyCore(plugin);
        }

        servicesManager = plugin.getServer().getServicesManager();
        servicesManager.register(Economy.class, economyProvider, plugin, ServicePriority.Normal);

        logger.info("Fully hooked into VaultAPI");
    }

    public void unhookVaultIntegration(){
        if(economyProvider != null){
            servicesManager = plugin.getServer().getServicesManager();
            servicesManager.unregister(Economy.class, economyProvider);

            logger.info("Unhooked from VaultAPI");
        }
    }
}

File: server/src/main/kotlin/io/github/eddiediamondfire/economyplus/dependency/EPVaultHook.kt
package io.github.eddiediamondfire.economyplus.dependency

import com.google.common.util.concurrent.ServiceManager
import io.github.eddiediamondfire.economyplus.Main
import io.github.eddiediamondfire.economyplus.utils.MessageManager
import net.milkbowl.vault.economy.Economy
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.plugin.RegisteredServiceProvider
import org.bukkit.plugin.ServicePriority
import org.bukkit.plugin.ServicesManager

class EPVaultHook(pluginClass: Main) {

    private val plugin: Main = pluginClass
    private val economyProvider: Economy = plugin.economyCore
    private val serviceManager: ServicesManager = Bukkit.getServicesManager()

    fun hookVault()
    {
        if(!vaultExist())
        {
            Bukkit.getPluginManager().disablePlugin(plugin)
            MessageManager.sendMessage(ChatColor.YELLOW, "Vault plugin is not found, disabling plugin!")
        }
        serviceManager.register(Economy::class.java, economyProvider, plugin, ServicePriority.Normal)
        MessageManager.sendMessage(ChatColor.GREEN, "Vault API is hooked into EconomyPlus")
    }

    fun unhookVault()
    {
        serviceManager.unregister(Economy::class.java, economyProvider)
        MessageManager.sendMessage(ChatColor.GREEN, "Vault API is unhooked from EconomyPlus")
    }
    private fun vaultExist(): Boolean{
        return plugin.server.pluginManager.getPlugin("Vault") != null
    }


}
